src/main/java/org/isf/menu/rest/UserSettingController.java

```diff
@@ -75,7 +80,16 @@ public UserSettingController(UserSettingMapper userSettingMapper, UserSettingMan
 	public ResponseEntity<UserSettingDTO> newUserSetting(@Valid @RequestBody UserSettingDTO userSettingDTO)
 					throws OHServiceException {
 		LOGGER.info("Attempting to create or update a UserSetting");
-		String userName = SecurityContextHolder.getContext().getAuthentication().getName();
+		String userName;
+		if (userSettingDTO.getUser() == null || userSettingDTO.getUser().equals("")) {
+			userName = SecurityContextHolder.getContext().getAuthentication().getName();
+		} else {
+			userName = userSettingDTO.getUser();
```

Comment:
why not just get the `userName` once before the first `if` instead and user that variable instead of going and getting it maybe 3 times?

Reply:
Yes, you're right.